src/signature.ts

```diff
@@ -44,17 +47,17 @@ function isValidSigRecovery(recovery: number): boolean {
  */
 export const ecrecover = function(
   msgHash: Buffer,
-  v: number,
+  v: number | BN | Buffer,
   r: Buffer,
   s: Buffer,
-  chainId?: number
+  chainId?: number | BN | Buffer
 ): Buffer {
   const signature = Buffer.concat([setLengthLeft(r, 32), setLengthLeft(s, 32)], 64)
   const recovery = calculateSigRecovery(v, chainId)
   if (!isValidSigRecovery(recovery)) {
     throw new Error('Invalid signature v value')
   }
-  const senderPubKey = ecdsaRecover(signature, recovery, msgHash)
+  const senderPubKey = ecdsaRecover(signature, recovery.toNumber(), msgHash)
```

Comment:
Note: recovery is a number between 0-3, see https://github.com/cryptocoinjs/secp256k1-node/blob/024fbdad3fb64499e8db8b35246c2f0a36afa8c8/lib/index.js#L294 (so we are fine to convert this to a number). I was a bit afraid that we would have to open an issue at secp256k1 to add BN/Buffer support for recovery ids.

Reply:
Do we then eventually want to use the [BNLike](https://github.com/ethereumjs/ethereumjs-util/blob/master/src/types.ts#L8) type to make these type of input formats consistent throughout the library (also for further additions)?

Along the same line:

1. Any reason to not add `Buffer` to the `BNLike` type, then we encompass all the functionality from here? 🤔 

2. Also: Would this be a candidate for a first `BigInt` introduction on the libraries, and so: can we also add a `BigInt` input type to `BNLike`? Or does this have side effects?

Reply:
I'd say that `BNLike` is of type `Buffer | BN | number | string | BigInt`. We can first start by expanding it with `Buffer`, and then on a seperate PR include `BigInt`s. I think it is very helpful if we add this flexible support in the libraries, it is especially helpful for our consumers. If they want a simple example, it doesn't make a lot of sense to convert the number `1` to a `BN`: this adds extra logic and an extra import. On the other hand, we do make sure our users are very strict and don't run into these rounding errors. But I would be very much in favor to add `BNLike` as argument type to some arguments, and this would definitely be the first library to start with.

Another side note: we might want to introduce this package to our monorepo. ( 😅  )

Reply:
Handling `BNLike` would be nice from a dev ux perspective, like we do in the `fromXData` factories in our monorepo packages. Maybe we can make some kind of utility/formatter to consume `BNLike` and output a specific type so it can be reused in other places to simplify the conversion logic.

> Another side note: we might want to introduce this package to our monorepo. ( 😅 )

yes I think it's a good time to do this, I have upgraded the config packages to v2 recently so @holgerd77 let me know if there's any other prep work for this repo before it would be eligible for merging into the monorepo.